server/src/internalClusterTest/java/org/elasticsearch/indices/recovery/IndexRecoveryIT.java

```diff
@@ -1048,9 +1063,7 @@ public void testRecoverLocallyUpToGlobalCheckpoint() throws Exception {
         final long lastSyncedGlobalCheckpoint = shard.getLastSyncedGlobalCheckpoint();
         final long localCheckpointOfSafeCommit;
         try (Engine.IndexCommitRef safeCommitRef = shard.acquireSafeIndexCommit()) {
-            localCheckpointOfSafeCommit = SequenceNumbers.loadSeqNoInfoFromLuceneCommit(
-                safeCommitRef.getIndexCommit().getUserData().entrySet()
-            ).localCheckpoint;
+            localCheckpointOfSafeCommit = getLocalCheckpointOfSafeCommit(safeCommitRef.getIndexCommit());
```

Comment:
I see the need for this now though I wonder if we could find a reasonable approximation and make the failing check below a range instead? We could use `[lcp, max]` as the range. This version is more precise though, so we can also take your route.

Reply:
I think we should keep this version to avoid possible false positives. The downside is that we're replicating production logic in the test case.